pkg/proxy/metrics/metrics.go

```diff
@@ -289,7 +289,7 @@ func RegisterMetrics(mode kubeproxyconfig.ProxyMode) {
 
 		switch mode {
 		case kubeproxyconfig.ProxyModeIPTables:
-			legacyregistry.CustomMustRegister(newCTStateInvalidPacketsCollector())
+			legacyregistry.CustomMustRegister(newNFAcctMetricCollector(IPTablesCTStateInvalidDroppedNFAcctCounter, iptablesCTStateInvalidDroppedPacketsDescription))
```

Comment:
could maybe have a top-level var

    iptablesCTStateInvalidDroppedPackets = newNFAcctMetricCollector(IPTablesCTStateInvalidDroppedNFAcctCounter, iptablesCTStateInvalidDroppedPacketsDescription)

so you can just do `legacyregistry.CustomMustRegister(iptablesCTStateInvalidDroppedPackets)` here?

Reply:
sure

Reply:
Should I define newNFAcctMetricCollector(...) where the metric description and counter name is defined instead? 
That will group all the metric-related objects.

```go
	// localhostNodePortsAcceptedPacketsDescription describe the metrics for the number of packets accepted
	// by iptables which were destined for nodeports on loopback interface.
	localhostNodePortsAcceptedPacketsDescription = metrics.NewDesc(
		"kubeproxy_iptables_localhost_nodeports_accepted_packets_total",
		"Number of packets accepted on nodeports of loopback interface",
		nil, nil, metrics.ALPHA, "")
	LocalhostNodePortAcceptedNFAcctCounter = "localhost_nps_accepted_pkts"
	localhostNodePortAcceptedMetricCollector = newNFAcctMetricCollector(...)
	
	
	
	
	legacyregistry.CustomMustRegister(localhostNodePortAcceptedMetricCollector)
	
```